infrastructure/adapters/persistence/src/main/java/md/maib/retail/infrastructure/persistence/EffectRecord.java

```diff
@@ -1,18 +1,33 @@
 package md.maib.retail.infrastructure.persistence;
 
 import lombok.AllArgsConstructor;
+import lombok.Getter;
 import md.maib.retail.model.effects.Effect;
 import md.maib.retail.model.effects.LoyaltyEffectType;
 
+import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
+
 @AllArgsConstructor
+@Getter
 public class EffectRecord {
     private UUID loyaltyEventTypeId;
-
     private String value;
+    private final LoyaltyEffectTypesAdapter loyaltyEffectTypesAdapter;
+    public Effect toEffect() {
+        LoyaltyEffectType loyaltyEffectType = loyaltyEffectTypesAdapter.findById(String.valueOf(loyaltyEventTypeId))
```

Comment:
Effect Record will be created by Hibernate (as proxy). So you need pass loyaltyEffectTypesAdapter as parameter of toEffect method